src/core/qgsdatasourceuri.h

```diff
@@ -168,6 +168,8 @@ class CORE_EXPORT QgsDataSourceURI
     QString database() const;
     //! Returns the port
     QString port() const;
+    //! Returns the driver
+    QString driver() const;
```

Comment:
Can you please add a setter for this as well, and corresponding tests in testqgsdatasourceuri.cpp?

It's also a good idea to mark the docstrings with the version members are added in, ie

```
//! Returns the driver
//! @note added in QGIS 2.16
```

This helps PyQGIS devs know which methods they can use in their plugins and scripts.